File: SDKs/amazonaws.com/ssm/2014-11-06/python/src/sdk/models/shared/opsfilter.py
from dataclasses import dataclass, field
from typing import List,Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class OpsFilter:
    r"""OpsFilter
    A filter for viewing OpsData summaries.
    """
    
    key: str = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('Key') }})
    values: List[str] = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('Values') }})
    type: Optional[OpsFilterOperatorTypeEnum] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Type') }})
    

File: SDKs/amazonaws.com/ssm/2014-11-06/typescript/src/sdk/models/shared/opsfilter.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { OpsFilterOperatorTypeEnum } from "./opsfilteroperatortypeenum";



// OpsFilter
/** 
 * A filter for viewing OpsData summaries.
**/
export class OpsFilter extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=Key" })
  key: string;

  @SpeakeasyMetadata({ data: "json, name=Type" })
  type?: OpsFilterOperatorTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=Values" })
  values: string[];
}
